internal/util/porch/function.go

```diff
@@ -50,81 +49,48 @@ func (f FunctionGetter) Get() []v1alpha1.Function {
 	codec := runtime.NewParameterCodec(scheme)
 	var functions v1alpha1.FunctionList
 	if err := client.Get().
-		Timeout(PorchExpiration).
+		Timeout(Expiration).
 		Resource("functions").
 		VersionedParams(&metav1.GetOptions{}, codec).
-		Do(f.ctx).
+		Do(f.Ctx).
 		Into(&functions); err != nil {
 		return nil
 	}
 	return functions.Items
 }
 
-type Matcher interface {
-	Match(v1alpha1.Function) bool
-}
-
-var _ Matcher = TypeMatcher{}
-var _ Matcher = KeywordsMatcher{}
-
-type TypeMatcher struct {
-	FnType string
-}
-
-// Match determines whether the `function` (which can be multi-typed), belongs
-// to the matcher's FnType. type value should only be `validator` or `mutator`.
-func (m TypeMatcher) Match(function v1alpha1.Function) bool {
-	for _, actualType := range function.Spec.FunctionTypes {
-		if string(actualType) == m.FnType {
-			return true
-		}
-	}
-	return false
-}
-
-type KeywordsMatcher struct {
-	Keywords []string
+// FunctionGetter gets a specific v1alpha1.Functions by name.
+type FunctionGetter struct {
+	Ctx context.Context
 }
 
-// Match determines whether the `function` has keywords which match the matcher's `Keywords`.
-// Experimental: This logic may change to only if all function keywords are found from  matcher's `Keywords`,
-// can it claims a match (return true).
-func (m KeywordsMatcher) Match(function v1alpha1.Function) bool {
-	if len(m.Keywords) == 0 {
-		// Accept all functions if keywords are not given.
-		return true
-	}
-	for _, actual := range function.Spec.Keywords {
-		for _, expected := range m.Keywords {
-			if actual != expected {
-				return true
-			}
-		}
+func (f FunctionGetter) Get(name string) (v1alpha1.Function, error) {
+	kubeflags := genericclioptions.NewConfigFlags(true)
+	var function v1alpha1.Function
+	client, err := CreateRESTClient(kubeflags)
+	if err != nil {
+		return function, err
 	}
-	return false
-}
-
-func MatchFunctions(ctx context.Context, matchers ...Matcher) []v1alpha1.Function {
-	var suggestedFunctions []v1alpha1.Function
-	functions := FunctionGetter{ctx: ctx}.Get()
-	for _, function := range functions {
-		for _, matcher := range matchers {
-			if matcher.Match(function) {
-				suggestedFunctions = append(suggestedFunctions, function)
-			}
-		}
+	scheme := runtime.NewScheme()
+	if err := v1alpha1.SchemeBuilder.AddToScheme(scheme); err != nil {
+		return function, err
 	}
-	return suggestedFunctions
+	codec := runtime.NewParameterCodec(scheme)
+	err = client.Get().
+		Timeout(Expiration).
+		Resource("functions").
+		Name(name).
+		Namespace("default").
```

Comment:
Get a resource by name requires providing the namespace. but "default" is not a reasonable assumption. 
Another option is to list all functions, which may not be efficient?
Do you have any insights what the right behavior should be if the functions have same repo, same function name but under different namespace? @martinmaly